src/ApacheModRewriteGenerator.php

```diff
@@ -63,20 +63,22 @@ public function generateRewrite( string $from, string $to, int $type ) : string
 			}
 		}
 
-		$output .= 'RewriteRule ^' . preg_quote(ltrim($fromPath, '/'), ' ') . '$ ' . $this->escapeSubstitution($prefix . ltrim($toPath, '/')) . '?' . $toQuery;
+		$output .= 'RewriteRule ^' . preg_quote(ltrim($fromPath, '/'), ' ') . '$ ' . $this->escapeSubstitution($prefix . ltrim($toPath, '/')) . $toQuery;
```

Comment:
> fix - no "?" added if no query string

The trailing question mark is actually very important and there for a reason. It tells Apache not to forward given GET parameters.

Reply:
But it adds a blank query string which messes up for a true "friendly URL rewriting".
If you do not add QSA in the rewrite rule options, no query string will be given.
My experiences with your tool show me that this is not necessary to end with "?"

Reply:
That is as designed.  It is designed to do _the exact rewrite the user enters_. Nothing more.

If you do not include a query string, a query string should not be included. If the user wants a query string, they can include a query string. Doing what the user told us to do is the more friendly option.

---

As you have it, the rewrite:

`foo.html?bar=10	foo.html`

Outputs:

```
RewriteCond %{QUERY_STRING} (^|&)bar\=10($|&)
RewriteRule ^foo\.html$ /foo.html [L,R=301]
```

Which results in an endless loop because the `?bar=10` is appended to the redirected URL.  

![image](https://user-images.githubusercontent.com/133747/71738209-95c60a00-2e1b-11ea-9a3c-58a8b1267403.png)

Reply:
You're right. I only used it without query string in "from" URL.
We can cancel this pull request ?
I'll work on my fork only.